portfolio/src/main/webapp/script.js

```diff
@@ -17,21 +17,35 @@
 
     *Description of functions below:
         *getData() asyncronously fetched data from github and calls functions to parse, filter and render component\
-        *SetHeader(repo) fetches and returns certain atributres from repo object
-        *SetRepos(repo) fetches and returns certain atributres from repo object
         *injectHeader(header) dynamically renders the header to the body
         *injectRepo(repo) creates a new repository elements and injects it into the body
+        *injectError() injects an error svg into the html file
 
 */
 
 async function getData (){
     await fetch('https://api.github.com/users/MohamedShatry/repos')
             .then(data => data.json())
             .then(res => {
-                let varHeader = SetHeader(res[0].owner);
+                if(res.length === 0){
+                    next();
+                }
+                let repo = res[0].owner;
+                let varHeader = {
+                    title : repo.login,
+                    url: repo.html_url,
+                    avatar_url: repo.avatar_url
+                }
                 injectHeader(varHeader);
+                
                 res.forEach(repo => {
-                    var returnRepo = SetRepos(repo);
+                    var returnRepo = {
```

Comment:
since this var is not being used after, you may just as well pass the object as argument directly. Same for varHeader.

Reply:
Please provide a reason why you wouldn't make the change before you resolve the comment. Please take care from next time.

Reply:
Hey Addarsh. I might have missed this comment. I apologize for that.
Will fix this going forward. 

Regards,

Reply:
No worries.